CSV enhancement

itroyano opened the pull request:
This PR addresses the following -
1) Adding an icon, description and links to the CSV for user-friendly output to be seen in operatorhub.io/preview
2) passing Preflight bundle check
3) typos and branding.

Fixes #43 

Signed-off-by: Igor Troyanovsky <itroyano@redhat.com>

itroyano commented:
Preview:

![image](https://user-images.githubusercontent.com/83590748/142453963-b0f22fcf-28f3-42ba-b62a-37ee2ada9f36.png)

itroyano commented:
/work-in-progress

acornett21 commented:
Nit: Remove the `new lines` that github is complaining about in a files. Other than that, I see no issues.

itroyano commented:
> Nit: Remove the `new lines` that github is complaining about in a files. Other than that, I see no issues.

done. thanks!

samira-barouti commented on bundle.Dockerfile:
```diff
@@ -39,5 +41,7 @@ LABEL quay.expires-after=${quay_expiration}
-COPY --from=builder /workspace/bundle/metadata /metadata/
+# Working around an issue where 'make bundle' will remove annotations from /metadata/annotations.yaml, causing a failed certification
+COPY --from=builder /workspace/annotationsWA /metadata/
+#COPY --from=builder /workspace/bundle/metadata /metadata/
```
can we remove this line?

itroyano replied:
done.

samira-barouti commented on bundle/metadata/annotations.yaml:
```diff
@@ -5,10 +5,14 @@ annotations:
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: operator-certification-operator
   operators.operatorframework.io.bundle.channels.v1: alpha
+  operators.operatorframework.io.bundle.channel.default.v1: alpha
```
Do we need this file as well? This is a duplicate of `annotationsWA/annotations.yaml`. Can we keep one of them?

itroyano replied:
removed it from PR.   I think it's a separate discussion if the file itself needs to be in the codebase.  @bcrochet has a thread on it.

bcrochet reviewed (commented):
Please explain what 'annotationsWA' means? Specifically, the 'WA' part? We need to have a longer discussion about what these things bring to us, and whether we should be doing these things in the first place.

itroyano commented:
@bcrochet 

> Please explain what 'annotationsWA' means? Specifically, the 'WA' part? We need to have a longer discussion about what these things bring to us, and whether we should be doing these things in the first place.

Preflight validates some annotations which `make bundle` doesn't generate. specifically `operators.operatorframework.io.bundle.channel.default.v1`

Moreover if I have an `annotations.yaml` which I've modified and added this annotation - running `make bundle` again with actually REMOVE the extra annotations.

As a WA, I'm keeping the annotations file static.  it doesn't have replaceable versions or operator-spec related attributes anyways.

bcrochet commented on bundle.Dockerfile:
```diff
@@ -27,6 +27,8 @@ LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=operator-certification-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
 LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.14.0
+LABEL operators.operatorframework.io.bundle.channels.v1: alpha
```
Are we sure this does the right thing? Can labels be specified as 'LABEL key: value'? The others are 'LABEL key=value'.

itroyano replied:
reverted

bcrochet reviewed (changes requested):
Check on dockerfile

bcrochet commented on bundle.Dockerfile:
```diff
@@ -39,5 +40,6 @@ LABEL quay.expires-after=${quay_expiration}
 # Copy files to locations specified by labels.
 COPY --from=builder /workspace/bundle/manifests /manifests/
-COPY --from=builder /workspace/bundle/metadata /metadata/
+# Working around an issue where 'make bundle' will remove annotations from /metadata/annotations.yaml, causing a failed certification
```
This should no longer be necessary.

itroyano replied:
reverted

itroyano commented:
Closing this PR in favor of the up to date consolidated #70
